apps/amo/decorators.py

```diff
@@ -238,3 +239,19 @@ def wrapper(request, username, *args, **kw):
             username = request.user.username
         return f(request, username, *args, **kw)
     return wrapper
+
+
+def atomic(fn):
+    """Set the transaction isolation level to SERIALIZABLE and then delegate
+    to transaction.atomic to run the specified code atomically. The
+    SERIALIZABLE level will run SELECTs in LOCK IN SHARE MODE when used in
+    conjunction with transaction.atomic.
+    Docs: https://dev.mysql.com/doc/refman/5.6/en/set-transaction.html.
+    """
+    @functools.wraps(fn)
+    def inner(*args, **kwargs):
+        cursor = connection.cursor()
+        cursor.execute('SET TRANSACTION ISOLATION LEVEL SERIALIZABLE')
```

Comment:
this hurts my brain a bit trying to think about master and slave connections. Is the cursor consistent (either master or slave) throughout the request? I don't even know what it would mean to start a transaction on a slave db that might rely on master changes. So perhaps this is something that should only be used when pinned to master anyway.